commands/uploader.go

```diff
@@ -223,7 +223,14 @@ func (c *uploadContext) prepareUpload(unfiltered ...*lfs.WrappedPointer) (*tq.Tr
 			c.trackedLocksMu.Lock()
 			c.unownedLocks = append(c.unownedLocks, lock)
 			c.trackedLocksMu.Unlock()
-			canUpload = false
+
+			// If the verification state is enabled or undefined,
+			// this failed locks verification means that the push
+			// should fail.
```

Comment:
This also doesn't fit the [table](https://github.com/git-lfs/git-lfs/pull/2168#discussion_r112969188), which states that pushes that aren't verified should just warn. Though I think being safer by default, and requiring the user to opt-out is a good move :+1: